optimizely/optimizely_user_context.py

```diff
@@ -46,7 +46,8 @@ def __init__(self, optimizely_client, user_id, user_attributes=None):
 
         self._user_attributes = user_attributes.copy() if user_attributes else {}
         self.lock = threading.Lock()
-        self.forced_decisions = {}
+        with self.lock:
+            self.forced_decisions = {}
```

Comment:
@ozayr-zaviar addition of thread safe locking mechanism looks good (applies for this whole commit). Thx for adding it.